api/src/wfl/module/covid.clj

```diff
@@ -207,17 +209,55 @@
+   as rows into `source-details-name` table."
+  [now dataset table row-ids]
+  (let [columns     (-> (datarepo/all-columns dataset table)
+                        (->> (map :name) set)
+                        (conj "datarepo_row_id"))
+        job-id (-> (datarepo/make-snapshot-request dataset columns table row-ids)
+                   (assoc :name now)
+                   (datarepo/create-snapshot-job))]
+    job-id))
+
 ;; Create and add new snapshots to the snapshot queue
-(defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now] [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
-          (update-last-checked [now])]
-    (let [now     (OffsetDateTime/now)
-          row-ids (find-new-rows source now)]
-      (when-not (empty? row-ids)
-        (write-snapshots source (make-snapshots source row-ids)))
-      (update-last-checked source now))))
+(defn ^:private update-tdr-source
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [{:keys [id dataset dataset_table table_column_name last_checked details] :as _source}]
+  (letfn [(find-new-rows [now]
+            (-> (datarepo/query-table-between
+                 dataset
+                 dataset_table
+                 table_column_name
+                 [last_checked now]
+                 [:datarepo_row_id])
+                :rows
+                flatten))
+          (make-snapshots [now-obj row-ids]
+            (let [shards (partition-all 500 row-ids)
+                  compact-now (.format now-obj (DateTimeFormatter/ofPattern "YYYYMMDD'T'HHMMSS"))]
+              (doall (map #(go-create-snapshot! compact-now dataset dataset_table %) shards))))
+          (write-snapshots-creation-jobs [tx now row-ids snapshots-creation-jobs]
+            (jdbc/insert-multi! tx
+                                details
+                                (map (fn [id] {:snapshot_creation_job_id id
+                                               :datarepo_row_ids         row-ids
+                                               :start_time               last_checked
+                                               :end_time                 now})
+                                     snapshots-creation-jobs)))
+          (update-last-checked [tx now]
+            (jdbc/update! tx (keyword tdr-source-table) {:last_checked now} ["id = ?" id]))]
+    (let [now-obj     (OffsetDateTime/now (ZoneId/of "UTC"))
+          now         (.format now-obj (DateTimeFormatter/ofPattern "yyyy-MM-dd HH:mm:ss"))
+          row-ids     (find-new-rows now)
+          tdr-job-ids (make-snapshots now-obj row-ids)]
+      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+        (write-snapshots-creation-jobs tx details now row-ids tdr-job-ids)
+        (update-last-checked tx now)))))
```

Comment:
Move the letfn into the let and this won't have to pass nows and row-ids around so much.
Or just break them out into separate defns like go-create-snapshot!.
Maybe stringify an #inst and truncate it instead?